src/types.ts

```diff
@@ -1,19 +1,28 @@
 const { isObject, isString, isMessageAttributeValid } = require('./validation');
 
-function entryFromObject(message) {
+interface Message {
+    id: string;
+    body: any;
+    groupId: string;
+    deduplicationId: string;
+    delaySeconds: number;
+    messageAttributes: object;
+}
+
+function entryFromObject(message: Message): any {
     if (!message.body) {
-        throw new Error("Object messages must have 'body' prop");
+        throw new Error(`Object messages must have 'body' prop`);
     }
 
     if (!message.groupId && !message.deduplicationId && !message.id) {
-        throw new Error("Object messages must have 'id' prop");
+        throw new Error(`Object messages must have 'id' prop`);
     }
 
     if (message.deduplicationId && !message.groupId) {
-        throw new Error("FIFO Queue messages must have 'groupId' prop");
+        throw new Error(`FIFO Queue messages must have 'groupId' prop`);
     }
 
-    const entry = {
+    const entry: any = {
```

Comment:
This looks like an AWS defined type, e.g `SendMessageBatchRequestEntry` (this needs confirming). Consider using predefined types where possible.